net/tcp/tcp_input.c

```diff
@@ -628,6 +628,37 @@ static void tcp_input(FAR struct net_driver_s *dev, uint8_t domain,
 
   dev->d_len -= (len + iplen);
 
+#if defined(CONFIG_NET_STATISTICS) && \
+    defined(CONFIG_NET_TCP_DEBUG_DROP_RECV)
+
+#pragma message \
```

Comment:
Maybe better change to `#warning`? I mean, do all compilers support pragma message?

Reply:
some boards treat warning as error, but the compilers do not provide other lower-level prints,

```
boards/risc-v/esp32c3/esp32c3-devkit/scripts/Make.defs:46:
CFLAGS := ... -Werror=return-type -Werror
```
vs(MSVC) and GCC support "pragma" expressions.

Reply:
I think `#warning` are not treated as errors. What about SDCC and pragma?

Reply:
Build break if enable -Werror

```
tcp/tcp_input.c: In function ‘tcp_input’:
tcp/tcp_input.c:634:2: error: #warning "CONFIG_NET_TCP_DEBUG_DROP_RECV is selected, this is debug " "feature to drop the tcp received packet on the floor, " "please confirm the configuration again if you do not want " "debug the TCP stack." [-Werror=cpp]
  634 | #warning \
      |  ^~~~~~~
```
I haven't tested SDCC, but #warning will bring errors when -Werror is enabled, which is wrong, and visual studio (MSVC) does not support #warning, I am considering whether it is possible replace all #warnings with #pragma message